src/widget/wcolorpickeraction.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <QHBoxLayout>
+#include <QWidget>
+#include <QWidgetAction>
+
+#include "widget/wcolorpicker.h"
+
+class WColorPickerAction : public QWidgetAction {
+  public:
+    WColorPickerAction(QWidget* parent)
+            : QWidgetAction(parent) {
+        m_pColorPicker = new WColorPicker(parent, true);
+
+        QHBoxLayout* pLayout = new QHBoxLayout();
+        pLayout->addWidget(m_pColorPicker);
+
+        QWidget* pWidget = new QWidget();
+        pWidget->setLayout(pLayout);
+        setDefaultWidget(pWidget);
+    }
+
+    ~WColorPickerAction() {
+        delete m_pColorPicker;
```

Comment:
If m_pColorPicker only lives as long as WColorPickerAction than `this` should be use as the parent and a parented_ptr is appropriate. Manual deletion in not necessary.